### ISSUE TITLE
How could I train ImageBind completely from scratch?

### ISSUE THREAD

**Issue by ChloeL19** [facebookresearch/ImageBind#51]
Thank you for the very cool work!
I'm having trouble finding your implementation of NCE loss, however. I know @fabawi has implemented a version of this for his LoRA fine-tuning version (kudos). However, if I wanted to train the original ImageBind model completely from scratch how would I do this?

**Comment by fabawi** [comment]
you can train the model without LoRA using [ImageBind-LoRA](https://github.com/fabawi/ImageBind-LoRA). Simply remove the --lora argument when calling train.py and set --full_model_checkpointing. I don't have the resources to fine-tune it but it should work in theory. Try it out on the toy dataset provided in the repo, and later you can try to implement dataloaders for the original datasets. 

On a side note, LoRA adaptation was shown to outperform full model fine-tuning in many instances and for different applications --- this is not specific to ImageBind-LoRA but a general statement about low-rank adaptation. If you would like to see the outcomes of fine-tuning on LoRA, check out [the example](https://github.com/fabawi/ImageBind-LoRA/example.py). The weights are in the repo itself. Just install the requirements and follow the instructions in the [README](https://github.com/fabawi/ImageBind-LoRA).

**Comment by Wolfwjs** [comment]
Hi @ChloeL19 ， May I ask if you have been able to train the model from scratch？

**Comment by wendellgithub0206** [comment]
I want to know how to train and how to apply the model after training is complete?
How to present the results?

**Comment by AI-ML-Enthusiast** [comment]
> you can train the model without LoRA using [ImageBind-LoRA](https://github.com/fabawi/ImageBind-LoRA). Simply remove the --lora argument when calling train.py and set --full_model_checkpointing. I don't have the resources to fine-tune it but it should work in theory. Try it out on the toy dataset provided in the repo, and later you can try to implement dataloaders for the original datasets.
> 
> On a side note, LoRA adaptation was shown to outperform full model fine-tuning in many instances and for different applications --- this is not specific to ImageBind-LoRA but a general statement about low-rank adaptation. If you would like to see the outcomes of fine-tuning on LoRA, check out [the example](https://github.com/fabawi/ImageBind-LoRA/blob/main/example.py). The weights are in the repo itself. Just install the requirements and follow the instructions in the [README](https://github.com/fabawi/ImageBind-LoRA).

Why use LoRA? Did ImageBind use it?
